test/client_ui_test.py

```diff
@@ -84,6 +85,74 @@ def test_draw():
     assert f.last == line.ljust(79) + "\n"
 
 
+def test_json_format():
+    fake_time = FakeTime()
+    f = FakeFile()
+
+    # Size is unknown at this point.
+    pb = client.ProgressBar(
+        phase="setting up", output=f, format="json", now=fake_time)
+    line_data = {
```

Comment:
Here we use line_data - which is only too verbose but wrong, since this is a dict.
Data should be used for binary data. Again using `progress` will probably better.